pkg/config/store.go

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/configmap"
+)
+
+type cfgKey struct{}
+
+const (
+	// PolicyControllerConfigName is the name of the configmap used to configure
+	// policy-controller.
+	PolicyControllerConfigName = "config-policy-controller"
+
+	// Specifies that if an image is not found to match any policy, it should
+	// be rejected.
+	DenyAll = "deny"
+
+	// Specifies that if an image is not found to match any policy, it should
+	// be allowed.
+	AllowAll = "allow"
+
+	WarnAll = "warn"
+
+	NoMatchPolicyKey = "no-match-policy"
+)
+
+// PolicyControllerConfig controls the behaviour of policy-controller that needs
+// to be more flexible than requiring a controller restart. Some examples are
+// controlling behaviour for what to do if no matching policies are found.
+// Point is that these apply to the whole controller instead of specific CIP
+// policies that apply only to matching images.
+type PolicyControllerConfig struct {
+	// NoMatchPolicy says what do in the case where an image does not match
+	// any policy.
+	NoMatchPolicy string `json:"no-match-policy"`
+}
+
+func NewPolicyControllerConfigFromMap(data map[string]string) (*PolicyControllerConfig, error) {
+	ret := &PolicyControllerConfig{NoMatchPolicy: "deny"}
+	switch data[NoMatchPolicyKey] {
+	case DenyAll:
+		ret.NoMatchPolicy = DenyAll
+	case AllowAll:
+		ret.NoMatchPolicy = AllowAll
+	case WarnAll:
+		ret.NoMatchPolicy = WarnAll
+	default:
+		ret.NoMatchPolicy = DenyAll
+	}
+	return ret, nil
+}
+
+func NewPolicyControllerConfigFromConfigMap(config *corev1.ConfigMap) (*PolicyControllerConfig, error) {
+	return NewPolicyControllerConfigFromMap(config.Data)
+}
+
+// FromContext extracts a PolicyControllerConfig from the provided context.
+func FromContext(ctx context.Context) *PolicyControllerConfig {
+	x, ok := ctx.Value(cfgKey{}).(*PolicyControllerConfig)
+	if ok {
+		return x
+	}
+	return nil
+}
+
+// FromContextOrDefaults is like FromContext, but when no
+// PolicyControllerConfig is attached, it returns a PolicyControllerConfig
+// populated with the defaults for each of the fields.
+func FromContextOrDefaults(ctx context.Context) *PolicyControllerConfig {
+	if cfg := FromContext(ctx); cfg != nil {
+		return cfg
+	}
+	return &PolicyControllerConfig{
+		NoMatchPolicy: DenyAll,
```

Comment:
Generally I'd vote for `deny` everything that doesn't match the policies, but I learnt that a blocking by-default behaviour might be too strict for new users.

Reply:
Yeah, that's why I made the configmap by default allow all to match current behaviour. However in case of any mistakes with the configmap, I want to fail closed by default..

Reply:
👍🏻